cukinia

```diff
@@ -621,6 +622,37 @@ _cukinia_mount()
 	return 0
 }
 
+# _cukinia_i2c: Check if i2c device exists
+# Optional: Check of a device address
+# arg1: the bus number
+# arg2: (optional) the device address
+_cukinia_i2c()
+{
+	local bus_number="$1"
+	local device_address="$2"
+	local ret
+
+	_cukinia_prepare "Checking if i2c-$bus_number exists"
+
+	test -d /sys/bus/i2c/devices/i2c-$bus_number
+	ret=$?
+
+	if [ $ret -eq 1 ]
+	then
+		return $ret
+	fi
+
+	if [ -n "$device_address" ]
```

Comment:
to keep the function leaner, we can reverse that test (if no device_address return, then continue with the i2c stuff)